### ISSUE TITLE
Incorrect complaint about the `_fields` attribute of instances of `typing.NamedTuple`

### ISSUE THREAD

**Issue by crazyscientist** [pylint-dev/pylint#9752]
### Bug description

Imagine this simplified example:

```python
from typing import Any, Dict, NamedTuple


class Example(NamedTuple):
    """Example docstring"""
    a: int
    b: str

    def as_dict(self) -> Dict[str, Any]:
        """Silly usage example for _fields attribute"""
        return {field: getattr(self, field) for field in self._fields}


example = Example(a=100, b="test")
print(example.as_dict())
```

It accesses the `_fields` attribute of `typing.NamedTuple`. This works as expected.

But pylint, complains.

### Configuration

I used the default configuration

### Command used

```shell
pylint <file_path>
```

### Pylint output

```python
************* Module file_path
<file_path>:11:57: E1101: Instance of 'Example' has no '_fields' member (no-member)
```

### Expected behavior

`NameTuple` provides the `_fields` attribute and pylint should not complain

### Pylint version

```shell
pylint 3.2.4
astroid 3.2.2
Python 3.10.12 (main, Nov 20 2023, 15:14:05) [GCC 11.4.0]
```

### OS / Environment

Ubuntu 22.04

**Comment by crazyscientist** [comment]
@mbyrnepr2 Yes, that one looks very similar, but I failed to find it before reporting.
